src/main/java/org/breedinginsight/brapi/v2/dao/BrAPIGermplasmDAO.java

```diff
@@ -190,48 +188,62 @@ private Map<String,BrAPIGermplasm> processGermplasmForDisplay(List<BrAPIGermplas
-                    } else if (additionalInfo.has(BrAPIAdditionalInfoFields.FEMALE_PARENT_UNKNOWN) && additionalInfo.get(BrAPIAdditionalInfoFields.FEMALE_PARENT_UNKNOWN).getAsBoolean()) {
-                        namePedigreeString = "Unknown";
+
+            // Get parent germplasm names without program key.
+            // This is designed so that pedigree="female/" will result in parentNames=["female", ""]
+            // and pedigree="/male" will result in parentNames=["", "male"];
+            // pedigree=null or pedigree="" will result in parentNames=[].
+            List<String> parentNames = new ArrayList<>();
+            if (pedigree != null) {
+                // Note: split with limit=-1 applies pattern as many times as possible, allowing capture of leading or trailing empty strings.
+                for (String name : pedigree.split("/", -1)) {
+                    if (!name.isEmpty())
+                    {
+                        // Strip program key.
+                        name = Utilities.removeProgramKeyAndUnknownAdditionalData(name, programKey);
                     }
+                    parentNames.add(name);
                 }
-                if (parents.size() == 2) {
-                    if (programGermplasmByFullName.containsKey(parents.get(1))) {
-                        String maleParentAccessionNumber = programGermplasmByFullName.get(parents.get(1)).getAccessionNumber();
-                        newPedigreeString += "/" + maleParentAccessionNumber;
-                        namePedigreeString += "/" + programGermplasmByFullName.get(parents.get(1)).getDefaultDisplayName();
-                        uuidPedigreeString += "/" + programGermplasmByFullName.get(parents.get(1)).getExternalReferences().
-                                stream().filter(ref -> ref.getReferenceSource().equals(referenceSource)).
-                                map(ref -> ref.getReferenceID()).findFirst().orElse("");
-                        additionalInfo.addProperty(BrAPIAdditionalInfoFields.GERMPLASM_MALE_PARENT_GID, maleParentAccessionNumber);
-                    }
+            }
+
+            // Update pedigree info for female parent.
+            if (parentNames.size() >= 1 && !parentNames.get(0).isEmpty())
+            {
+                gidPedigreeString = additionalInfo.has(GERMPLASM_FEMALE_PARENT_GID) ? additionalInfo.get(GERMPLASM_FEMALE_PARENT_GID).getAsString() : "";
+                namePedigreeString = parentNames.get(0);
+                uuidPedigreeString = additionalInfo.has(GERMPLASM_FEMALE_PARENT_UUID) ? additionalInfo.get(GERMPLASM_FEMALE_PARENT_UUID).getAsString() : "";
+                // Throw a descriptive error if femaleParentUUID is absent.
+                if (!additionalInfo.has(GERMPLASM_FEMALE_PARENT_UUID)) {
+                    log.debug("The germplasm data in your backing BrAPI service needs to be updated: https://github.com/Breeding-Insight/bi-api/pull/290");
```

Comment:
It will be more helpful to have the log message reference the program in error instead of "your"

Reply:
Done.